src/components/ha-date-range-picker.ts

```diff
@@ -317,6 +331,28 @@ export class HaDateRangePicker extends LitElement {
     `;
   }
 
+  private _handleNext(): void {
+    const diff = this.endDate.getTime() - this.startDate.getTime();
+    const dateRange = [
+      new Date(this.endDate.getTime() + 1),
+      new Date(this.endDate.getTime() + 1 + diff),
+    ];
+    const dateRangePicker = this._dateRangePicker;
+    dateRangePicker.clickRange(dateRange);
+    dateRangePicker.clickedApply();
+  }
+
+  private _handlePrev(): void {
+    const diff = this.endDate.getTime() - this.startDate.getTime() + 1;
+    const dateRange = [
+      new Date(this.startDate.getTime() - diff),
+      new Date(this.startDate.getTime() - 1),
+    ];
+    const dateRangePicker = this._dateRangePicker;
+    dateRangePicker.clickRange(dateRange);
+    dateRangePicker.clickedApply();
+  }
+
   private _setDateRange(ev: CustomEvent<ActionDetail>) {
```

Comment:
Please use `date-fns` to calculate the time.

I understand your +/- 1, technically it is correct but I wouldn't go into that detail:
- 2pm - 5pm -> 11am - 2pm
Is way easier to read for the user

Reply:
I will change the date-conversion using date-fns.
Yes, the reason for +1 and -1 is that the time range is diplayed 0:00 to 23:59 for today, this month, week etc. The End Date is missing 1 millisecond.
So, if I use the real timespan to shift Previous multiple times it does not move exact days.